src/tribler/gui/core_manager.py

```diff
@@ -118,9 +130,44 @@ def start_tribler_core(self):
         self.core_process.start(sys.executable, core_args)
 
     def on_core_started(self):
+        self._logger.info("Core process started")
         self.core_started = True
+        self.core_started_at = time.time()
         self.core_running = True
-        self.events_manager.connect(reschedule_on_err=True)  # retry until REST API is ready
+        self.check_core_api_port()
+
+    def check_core_api_port(self, *args):
+        """
+        Determines the actual REST API port of the Core process.
+
+        This function is first executed from the `on_core_started` after the physical Core process starts and then
+        repeatedly executed after API_PORT_CHECK_INTERVAL seconds until it retrieves the REST API port value from
```

Comment:
"after `API_PORT_CHECK_INTERVAL` seconds" --> to avoid any confusion, this is in milliseconds

Reply:
Yep, thanks!